runtime/connectors/s3/s3.go

```diff
@@ -76,3 +87,66 @@ type connector struct{}
 func (c connector) Spec() connectors.Spec {
 	return spec
 }
+
+// Consume will eventually be added to the interface
+func Consume(ctx context.Context, source *connectors.Source) (string, error) {
+	conf, _ := ParseConfig(source.Properties)
+
+	// The session the S3 Downloader will use
+	sess := session.Must(session.NewSession(&aws.Config{
+		Region:      &conf.AWSRegion,
+		Credentials: getAwsCredentials(conf),
+	}))
+
+	// Create a downloader with the session and default options
+	downloader := s3manager.NewDownloader(sess)
+
+	bucket, key, filename, err := getAwsUrlParts(conf.Path)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse path %s, %v", conf.Path, err)
+	}
+
+	filename = os.ExpandEnv(fmt.Sprintf("$TMPDIR%s", filename))
+	// Create a file to write the S3 Object contents to.
+	f, err := os.Create(filename)
+	if err != nil {
+		return "", fmt.Errorf("failed to create file %q, %v", filename, err)
+	}
+
+	// Write the contents of S3 Object to the file
+	n, err := downloader.Download(f, &s3.GetObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to download file, %v", err)
+	}
+	fmt.Printf("file downloaded, %d bytes\n", n)
+
+	return filename, nil
+}
+
+func getAwsCredentials(conf *Config) *credentials.Credentials {
+	if conf.AWSSession != "" {
+		return credentials.NewStaticCredentialsFromCreds(credentials.Value{
+			SessionToken: conf.AWSSession,
+		})
+	} else if conf.AWSKey != "" && conf.AWSSecret != "" {
+		return credentials.NewStaticCredentialsFromCreds(credentials.Value{
+			AccessKeyID:     conf.AWSKey,
+			SecretAccessKey: conf.AWSSecret,
+		})
+	}
+	return nil
```

Comment:
I assume returning nil will cause it to use system default credentials? (From `~/.aws` or something?)

Reply:
Yes that is correct.